src/Dotnet.Installer.Core/Models/Component.cs

```diff
@@ -222,9 +222,35 @@ public async Task Unmount(IFileService fileService, IManifestService manifestSer
         logger?.LogDebug("Removed empty directories.");
     }
 
-    public DotnetVersion GetDotnetVersion(IManifestService manifestService, IFileService fileService)
+    public async Task<DotnetVersion> GetRemoteDotnetVersion(
```

Comment:
**suggestion (non-blocking):** We can gather both local and remote .NET versions that same way.

The `GetLocalDotnetVersion` function relies on the `.version` file present in the .NET component directories. But since you already implemented a Snapd REST API integration, we can also use the `GET /v2/snaps` endpoint to query snaps installed in the system and parse their versions. Therefore, we can use a single function that branches off based on whether `Component.Installation` is `null` or not, for example.

There's also the added benefit that we don't have to rely on a file that we don't control (`.version`) and might stop existing in the future without proper heads-up.

Reply:
Just to write this down publicly: Getting the version of an installed non-root component is not easily possible, because the component is not installed as a separate snap, because the component is provided via an installed root component. Therefore we are not able to get the version of the installed non-root component via snapd. We could query the version in the snap store, but there is no guarantee that this is the installed version.

After discussing this issue for a while we decided to get rid of root/non-root components. Every component will provide it's content via a separate content snap. This allows us in the future to use snapd to get the version of all installed components.

In the mean time we will continue using the file service to get the version of installed non-root components.